src-mobile/script/components/putFade.js

```diff
@@ -0,0 +1,38 @@
+export function putFade(classFade, carouselName, duration) {
+  const carousels = document.querySelectorAll(`.${carouselName}`);
+  const animationDuration = duration;
+
+  let animationStartTime = null;
+
+  // Function to toggle the "classFade" class based on animation progress
+  const toggleTrackNameFade = (progress) => {
+    carousels.forEach((carousel) => {
+      if (progress >= 0 && progress <= 20) {
+        carousel.parentElement.classList.remove(`${classFade}`);
+      } else {
+        carousel.parentElement.classList.add(`${classFade}`);
+      }
+    });
+  };
+
+  // Function to update the carousel based on current time and animation duration
+  const updateCarousel = () => {
+    const currentTime = Date.now();
+    const elapsedTime = (currentTime - animationStartTime) % animationDuration;
```

Comment:
why are you getting the module of this division?

Reply:
the updateCarousel is calculating the elapsed time since the start of the animation. We achieve this by subtracting `animationStartTime` (the time when the animation began) from `currentTime` (the current time). The module operator is used to obtain the remainder of the division by `animationDuration`, ensuring that `elapsedTime` always falls within the range of 0 to `animationDuration`.